### DISCUSSION TITLE
[openssl][build] DESTDIR and OPENSSLDIR usage difference between unix and Windows

### DISCUSSION THREAD

**Discussion by kulkarniamit** [openssl/openssl#22324] [Q&A]
### Building and installing openssl 3 (OpenSSL 3.0/3.1) on Unix
```
$ ./Configure --prefix=/tmp/ossl3 --openssldir=/ssldir no-zlib no-shared
Configuring OpenSSL version 3.0.12-dev for target linux-x86_64
...
$ make
...
$ make DESTDIR=/tmp/installer OPENSSLDIR=/confs install_ssldirs
created directory `/tmp/installer'
created directory `/tmp/installer/confs'
created directory `/tmp/installer/confs/certs'
created directory `/tmp/installer/confs/private'
created directory `/tmp/installer/confs/misc'
...
install ./apps/openssl.cnf -> /tmp/installer/confs/openssl.cnf
...
$ ls  /tmp/installer/confs/openssl.cnf
/tmp/installer/confs/openssl.cnf
```
* On Unix platform, `DESTDIR` and `OPENSSLDIR` are [concatenated](https://github.com/openssl/openssl/blob/master/Configurations/unix-Makefile.tmpl#L701) (when specified) to create the final target path for copying configuration files. This behavior doesn't appear to be consistent between Unix and Windows.

### Building and installing openssl 3 (OpenSSL 3.0/3.1) on Windows
```
C:\> perl Configure --prefix=C:\tmp\ossl-install-dir --openssldir=ssldir
...
C:\> nmake
...
C:\>nmake DESTDIR=C:\tmp\test\ OPENSSLDIR=\random install_ssldirs

Microsoft (R) Program Maintenance Utility Version 14.37.32824.0
Copyright (C) Microsoft Corporation.  All rights reserved.

created directory `/random'
created directory `/random/certs'
created directory `/random/private'
created directory `/random/misc'
Copying: ./apps/openssl.cnf to /random/openssl.cnf.dist
Copying: ./apps/openssl.cnf to /random/openssl.cnf
...
C:> dir /b C:\random\openssl.cnf
openssl.cnf
```

* On Windows, [OPENSSLDIR](https://github.com/openssl/openssl/blob/master/Configurations/windows-makefile.tmpl#L532) is used as prefix for copying configuration files.

Why doesn't the installation command on Windows install `openssl.cnf` in `C:\tmp\test\random` (`$(DESTDIR)/$(OPENSSLDIR)`) like Unix? Is it expected that `DESTDIR` will be ignored when `OPENSSLDIR` is specified on Windows?

_NOTE: `DESTDIR` is used as prefix when `OPENSSLDIR` is not specified on Windows._

**Comment by levitte** [comment]
That was unexpected, I think... I'll have a closer look.
